File: dotnet/utils/SecretLoader.cs
using System.Text.Json;

namespace QAItest.Playwright.DotNet.Utils;

public static class SecretLoader
{
    public static Dictionary<string, object> GetSecretMap(string secretId)
    {
        if (secretId.StartsWith("env:"))
        {
            var envName = secretId[4..];
            return Parse(Environment.GetEnvironmentVariable(envName));
        }

        if (secretId.StartsWith("file:"))
        {
            var filePath = Path.GetFullPath(secretId[5..]);
            return File.Exists(filePath) ? Parse(File.ReadAllText(filePath)) : new Dictionary<string, object>();
        }

        var envValue = Environment.GetEnvironmentVariable(secretId);
        if (!string.IsNullOrWhiteSpace(envValue))
        {
            return Parse(envValue);
        }

        var candidatePath = Path.GetFullPath(secretId);
        return File.Exists(candidatePath) ? Parse(File.ReadAllText(candidatePath)) : new Dictionary<string, object>();
    }

    private static Dictionary<string, object> Parse(string? value)
    {
        if (string.IsNullOrWhiteSpace(value))
        {
            return new Dictionary<string, object>();
        }

        return JsonSerializer.Deserialize<Dictionary<string, object>>(value) ?? new Dictionary<string, object>();
    }
}

File: java/utils/SecretLoader.java
package utils;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Collections;
import java.util.Map;

public final class SecretLoader {
  private static final ObjectMapper MAPPER = new ObjectMapper();

  private SecretLoader() {}

  public static Map<String, Object> getSecretMap(String secretId) throws IOException {
    if (secretId.startsWith("env:")) {
      String env = System.getenv(secretId.substring(4));
      return env == null || env.isBlank() ? Collections.emptyMap() : MAPPER.readValue(env, Map.class);
    }
    if (secretId.startsWith("file:")) {
      Path path = Path.of(secretId.substring(5)).toAbsolutePath();
      return Files.exists(path) ? MAPPER.readValue(Files.readString(path), Map.class) : Collections.emptyMap();
    }
    String env = System.getenv(secretId);
    if (env != null && !env.isBlank()) {
      return MAPPER.readValue(env, Map.class);
    }
    Path path = Path.of(secretId).toAbsolutePath();
    return Files.exists(path) ? MAPPER.readValue(Files.readString(path), Map.class) : Collections.emptyMap();
  }
}
